quetz/tasks/mirror.py

```diff
@@ -404,27 +444,46 @@ def handle_batch(update_batch):
 
             return False
 
-        for package_name, metadata in packages.items():
-            if check_package_membership(package_name, includelist, excludelist):
-                path = os.path.join(arch, package_name)
-
+        # go through all packages from remote channel
+        channel_metadata = channel.load_channel_metadata()
+        for repo_package_name, metadata in packages.items():
+            action = check_package_membership(
+                channel,
+                channel_metadata,
+                repo_package_name,
+                metadata,
+                remote_host=remote_repository.host,
+            )
+            if action == MembershipAction.INCLUDE:
                 # try to find out whether it's a new package version
-
                 is_uptodate = None
                 for _check in version_checks:
-                    is_uptodate = _check(package_name, metadata)
+                    is_uptodate = _check(repo_package_name, metadata)
```

Comment:
Why `repo_package_name` instead of `package_name`?

Reply:
I think the latter is a bit clearer about that it's something like `numpy-1.23.4-py39hefdcf20_0.tar.bz2` and not just `numpy`